Library/Homebrew/cmd/info.rb

```diff
@@ -54,10 +55,15 @@ def print_info
           else
             info_formula Formulary.find_with_priority(f)
           end
-        rescue FormulaUnavailableError
+        rescue FormulaUnavailableError => e
           # No formula with this name, try a blacklist lookup
-          raise unless (blacklist = blacklisted?(f))
-          puts blacklist
+          if (blacklist = blacklisted?(f))
+            ofail "#{e.message}\n#{blacklist}"
+          else
+            ofail e.message
+            ohai "Searching for migrated formulae..."
```

Comment:
It doesn't take long, could be left out. I put it in just so that we print a message for each kind of search.